### PR TITLE
[CSV-304] Accessors for header/trailer comments

### PR DESCRIPTION
**Pull request by pedro-w**
Add accessors for header comments (before the header row)
and trailer comments (after the last record)
Also add javadoc and tests

See https://issues.apache.org/jira/browse/CSV-304

### CODE REVIEW

**Review by kinow — COMMENTED** [review]
One comment, but the rest of the code looks good! Thanks @pedro-w!

**Comment by pedro-w** [comment]
I noticed that github is parsing the Javadoc annotation `@since` as a username. I apologize to the owner of that account.

**Review by kinow — APPROVED** [review]
Looks good to me. Asked a review from @garydgregory as he works on and knows a lot more about this component. Thanks @pedro-w !

File: src/test/java/org/apache/commons/csv/CSVParserTest.java [hunk 1]
```diff
@@ -1375,4 +1375,147 @@ private void validateRecordPosition(final String lineSeparator) throws IOExcepti
...
+    }
+    @Test
+    public void testGetTrailerComment_HeaderComment3() throws IOException {
+        try (CSVParser parser = CSVParser.parse(CSV_INPUT_HEADER_COMMENT, FORMAT_EXPLICIT_HEADER_NOSKIP)) {
+            parser.getRecords();
+            assertFalse(parser.hasTrailerComment());
+            assertNull(parser.getTrailerComment());
+        }
+    }
+    @Test
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/csv/CSVParserTest.java:1512]: Please separate methods with a blank line.

File: src/main/java/org/apache/commons/csv/CSVParser.java [hunk 2]
```diff
@@ -713,6 +764,10 @@ CSVRecord nextRecord() throws IOException {
             case EOF:
                 if (this.reusableToken.isReady) {
                     this.addRecordValue(true);
+                } else {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/csv/CSVParser.java:767]: Extra block level is not needed, IOW use "else if" instead of "else { if ... }"

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @pedro-w,
Is the intent to allow comments at the start and end of a file or before and after each record?
Please see my scattered comments as well.

**Comment by pedro-w** [comment]
> 


> Is the intent to allow comments at the start and end of a file or before and after each record?

The former. I'll explain my thinking, *please correct me if I've got it wrong!*

    ; Comment 0
    A,B
    ; Comment 1
    1,1
    ; Comment 2
    2,2
    ; Comment 3

The comments attach to the following line so Comments 1 & 2 can currently be extracted from the corresponding `CSVRecord`s and Comment 3 can never be accessed. Comment 0 is a special case. If we ask `CSVParser` to generate the column headings from the first record, it can't be accessed, but otherwise it's attached to "A,B" which is just a normal data record.

My intention was to allow access to Comment 0 via `getHeaderComment` (only in the case of generated column names) and Comment 3 via `getTrailerComment` (always).


I'll get onto the other issues too as they're mostly just formatting.

**Pull request merged**